src/components/Wizard/CreateVmWizard/providers/VMWarePasswordAndCheck.js

```diff
@@ -0,0 +1,212 @@
+  }
+
+  if (phase === PhaseLoadingVmDetailFailed) {
+    return <ReadVmDetailFailed />;
+  }
+
+  // eslint-disable-next-line no-console
+  console.warn('VMWareProviderStatusByPhase unrecognized phase found within the V2V VMWare object: ', phase);
+  return null;
+};
+VMWareProviderStatusByPhase.defaultProps = {
+  phase: undefined,
+};
+VMWareProviderStatusByPhase.propTypes = {
+  phase: PropTypes.string,
+};
+
+// see onVmwareCheckConnection() for details
+const VMWareProviderStatus = ({ connValue, extraProps }) => {
+  if (!connValue) {
+    return null;
+  }
+  const { providerStatus, V2VVmwareName } = connValue;
+
+  if (!V2VVmwareName) {
+    return null;
+  }
+
+  const { WithResources, basicSettings } = extraProps;
+
+  if (providerStatus === PROVIDER_STATUS_CONNECTION_FAILED) {
+    return <ConnectionFailedInfra />;
+  }
+
+  if (providerStatus === PROVIDER_STATUS_CONNECTING && !V2VVmwareName) {
+    return <CheckingCredentials />;
+  }
+
+  const resourceMap = {
+    v2vvmware: {
+      resource: getResource(V2VVMwareModel, {
+        name: V2VVmwareName,
+        namespace: get(basicSettings[NAMESPACE_KEY], 'value'),
+        isList: false,
+      }),
+    },
+  };
+  const resourceToProps = ({ v2vvmware }) => ({
+    // TODO: handle the case when V2VVmwareName is set but the object is missing in API (404)
+    // means: it was garbage collected (so rare case)
+    phase: get(v2vvmware, 'status.phase'), // value set by the controller
+  });
+
+  return (
+    <WithResources resourceMap={resourceMap} resourceToProps={resourceToProps}>
+      <VMWareProviderStatusByPhase />
+    </WithResources>
+  );
+};
+VMWareProviderStatus.defaultProps = {
+  connValue: null,
+};
+VMWareProviderStatus.propTypes = {
+  extraProps: PropTypes.object.isRequired,
+  connValue: PropTypes.object,
+};
+
+// workaround to wrap two components at a single row
+const VMWarePasswordAndCheck = ({ onChange, id, value, extraProps }) => {
+  const { onCheckConnection } = extraProps;
+  const pwdValue = get(value, PROVIDER_VMWARE_USER_PWD_KEY);
+  const connValue = get(value, PROVIDER_VMWARE_CONNECTION);
+
+  return (
+    <Fragment>
+      <div className="kubevirt-create-vm-wizard__import-vmware-passwordcheck">
+        <Text
+          type={PASSWORD}
+          id={`${id}-text`}
+          value={pwdValue || ''}
+          onChange={newValue =>
+            onChange({
+              [PROVIDER_VMWARE_USER_PWD_KEY]: newValue,
+              [PROVIDER_VMWARE_CONNECTION]: connValue,
+            })
+          }
+        />
+        <Button
+          id={`${id}-check-button`}
+          className="kubevirt-create-vm-wizard__import-vmware-passwordcheck-button"
+          onClick={() => {
+            onCheckConnection(newValue =>
+              onChange({
+                [PROVIDER_VMWARE_USER_PWD_KEY]: pwdValue,
+                [PROVIDER_VMWARE_CONNECTION]: newValue,
+              })
+            );
+          }}
+        >
+          Check
```

Comment:
This is per-design, the button is small for longer text.
What about keeping the layout as it is designed atm but changing the button text to `Login`?

WDYT, @lizsurette ? 
The question is about text/position of the `Check` button in the "Create New Vmware Connection" sub-part of the dialog.

Once the user hits the button, there's attempt to connect to the VMWare which is immediately followed by retrieving of the VMS list. The user watches progress/failure as designed (meens text/alert bellow the button).
In case of success, the next dropdown is populated with the names of VMs.

Reply:
@mareklibra I think "Login" works! I think @andybraren and I went back and forth on the right term here so I don't see any reason why that couldn't be our starting point and then we can get some feedback from users.

Reply:
I'd say our options are either: `Check`, `Verify`, `Validate`, `Log in`, or `Connect`.

I agree with @lizsurette that user feedback should determine what we end up using. My best guess is that they will be fine with `Check` because it is:

1) More common than `Verify`
2) More casual than `Validate`
3) More accurate than `Log in` (I feel like this is an unusual context to see "Log in")
4) Less technical than `Connect`

But I'm not passionate about this. I'm much more concerned about @rawagner's point that the button may not be obvious enough. I've mocked up two options.

![option-1](https://user-images.githubusercontent.com/9122899/53273604-7110ea00-36c2-11e9-8522-c9473855f77d.png)

Option 1 puts the Check button below the Password field as @rawagner suggested.

Behavior-wise, the button can stay enabled even while the fields are blank (one less dynamic thing to worry about) since any empty fields would be called out via error text. None of these fields or buttons should disappear after the check completes (the user may decide to change their credentials). The "Remember vCenter credentials" checkbox below the button becomes enabled if the credentials are valid, and unchecked/disabled if the user modifies any of the relevant fields after a successful check (same goes for the "VM to import" dropdown).

![option-2](https://user-images.githubusercontent.com/9122899/53273808-001e0200-36c3-11e9-926f-9df4ea88dac2.png)

Option 2 is just making the Check button a primary blue color, with the same behavior described above.

I vote Option 1 (thoughts @lizsurette?)

Reply:
> I vote Option 1 (thoughts @lizsurette?)

+1 to option 1. Thanks for thinking these options through @andybraren !

Reply:
Thanks for the ideas. I've rearranged the password and Check-button based on option 1.
The Check button is disabled when the URL is not provided (username or pwd can be still empty).

I will implement the disabling of the checkbox in a later PR, probably new asynchronous flow will be needed for that. Please let me know if you think otherwise.